### ISSUE TITLE
San d'Oria Mission 1-3 "Save the Children"

### ISSUE THREAD

**Issue by bigpurpleharness** [Demiurge-DSP/feedback-and-info#34]
**Feedback or issue :**  

When repeating Sandy Mission 1-3 "Save the Children" you do not obtain a key item after finishing the BCNM. When performing this mission for the first time, it works correctly. Upon repeating, you get the option to fight in the BCNM, however when finish and checking the hut again for the key item, it simply offers the fight to you again. Speaking to the town guard warrants the, "Go to Ghelsba" speech as if you have just accepted the mission. 

- [x] **I understand that I should fill in with an x or check these boxes for yes/no**  

- [x] **I understand that using the AT `@` symbol tells github to notify a user of that name**  

- [x] **I understand that I** **_should not_** **use this form to report missing features or general bugs with Darkstar**  

- [x] **I understand that if this is reporting a bug specific to this server I need to be very detailed, including info such as but not limited to my position and the date and time of occurrence, any NPC or items involved, etc. etc.**

**Comment by TeoTwawki** [comment]
Did you get a cs event at Arnau in Northern San d'Oria during the repeat of the mission?

**Comment by TeoTwawki** [comment]
I also need the character name that is not getting the key item.

**Comment by bigpurpleharness** [comment]
Character name is Alaik and I tried doing that although its said its not needed. It does let you progress past that point as you can enter and defeat the orcs.

**Comment by TeoTwawki** [comment]
Looks like it will only give the KI if the mission status variable is set to 2, and that NPC is the only place I see it being set to 2. Is your character available in game tonight? if not I will try and catch you on discord tomorrow.

**Comment by bigpurpleharness** [comment]
I'll be on around 7pm CST tonight, I'll retry talking to all of the NPCs again and let you know.

**Comment by bigpurpleharness** [comment]
Okay so some odd things happened. I rechecked all NPCs, and when getting to Arnau I got the mission completion cutscene (Where he thanks you for completing the mission, not where he tells you what's going on.)

I rechecked the guard after thinking maybe that was all that's needed to flag it as finished with no luck. Attempting to perform the fight again after getting the completion cutscene awarded the KI. 

So in short, the mission is done, and I have no idea what was going on.

**Comment by TeoTwawki** [comment]
I made an alteration to the BCNM end scene but I was trying to catch you to check your variable state before you went back in. We never caught up with each other so I was unable to do that.

Arneu has multiple related scenes - one can only play AFTER you've cleared the BCNM, its story/flavor test stuff. The other is intended to be seen During the mission before the BCNM. 

Then when you repeat the mission, the code allows you to go straight to the BCNM without seeing him again even though its expecting a variable set by him... So the fact you got the "after cs" hints that your variable never set, you basically needed to see him multiple times there to proceed "normally". The half-fix I pushed makes the bcnm not care - if you can enter, you can win, as opposed to entering repeatedly without credit.

To fully fix the entire mission need re-evaluated to account for the after cs and the correct repeat behaviors.
